zaps/0002-encrypt-push-notifications.md

```diff
@@ -0,0 +1,182 @@
+- ZAP PR: [zulip/zulip-architecture#2](https://github.com/zulip/zulip-architecture/pull/2)
+
+# Summary
+
+Proposal to add end-to-end encryption for mobile push notifications.
+
+# Motivation
+
+Currently, the Zulip Push Notification Forwarding Service receives
+(but does not store or log) the contents of individual mobile push notifications.
+
+Adding end-to-end encryption for push notifications will ensure that the
+actual message content does not need to be readable by the push notification
+bouncer (or Google/Apple's servers, for that matter).
+
+# Detailed design
+
+## High-Level Plan
+
+- Client checks the server if it supports E2E encryption. If it doesn't,
+  use the legacy push notification registration flow.
+
+- Client makes an API call to the server to register itself for push notifications,
+  receives a shared secret key in response.
+
+- While sending notification, the server encrypts the notification content
+  and sends it to push bouncer. Push bouncer forwards it to FCM/APNs.
```

Comment:
clarify it's encrypted using the shared secret from above

Reply:
The above comment is outdated.

Updated the text to align with our plan to use public-key encryption.